bindings/pyroot_experimental/PyROOT/src/CPPInstancePyz.cxx

```diff
@@ -0,0 +1,123 @@
+// Author: Massimiliano Galli CERN  07/2019
+// Original PyROOT code by Wim Lavrijsen, LBL
+
+/*************************************************************************
+ * Copyright (C) 1995-2018, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+// Bindings
+#include <iostream>
+#include "CPyCppyy.h"
+#include "PyROOTPythonize.h"
+#include "CPPInstance.h"
+#include "ProxyWrappers.h"
+#include "Converters.h"
+#include "Utility.h"
+#include "PyzCppHelpers.hxx"
+#include "TBufferFile.h"
+#include "CustomPyTypes.h"
+
+using namespace CPyCppyy;
+
+namespace PyROOT {
+    extern PyObject* gRootModule;
+}
+
+/// This method is a helper for (un)pickling of ObjectProxy instances.
+PyObject* PyROOT::CPPInstanceExpand( PyObject* /*self*/, PyObject* args ) {
+    PyObject* pybuf = 0, *pyname = 0;
+    if (!PyArg_ParseTuple(args, const_cast< char* >( "O!O!:__expand__" ), &PyBytes_Type, &pybuf, &PyBytes_Type, &pyname))
+        return 0;
+    const char* clname = PyBytes_AS_STRING(pyname);
+    // TBuffer and its derived classes can't write themselves, but can be created
+    // directly from the buffer, so handle them in a special case
+    void* newObj = 0;
+    if (strcmp(clname, "TBufferFile" ) == 0) {
+        TBufferFile* buf = new TBufferFile( TBuffer::kWrite );
+        buf->WriteFastArray(PyBytes_AS_STRING(pybuf), PyBytes_GET_SIZE(pybuf));
+        newObj = buf;
+    } else {
+        // use the PyString macro's to by-pass error checking; do not adopt the buffer,
+        // as the local TBufferFile can go out of scope (there is no copying)
+        TBufferFile buf(TBuffer::kRead, PyBytes_GET_SIZE(pybuf), PyBytes_AS_STRING(pybuf), kFALSE );
+        newObj = buf.ReadObjectAny(0);
+    }
+    PyObject* result = BindCppObject(newObj, Cppyy::GetScope(clname));
+    if (result) {
+        // this object is to be owned by the interpreter, assuming that the call
+        // originated from there
+        ((CPPInstance*)result)->PythonOwns();
+    }
+    return result;
+}
+
+/// PyROOT object proxy pickle support
+PyObject* op_reduce(CPPInstance* self) {
+    // Turn the object proxy instance into a character stream and return for
```

Comment:
This three lines of comment are the documentation of what the function does, I would put them right above the function.